sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosClientBuilder.java

```diff
@@ -327,6 +317,46 @@ public CosmosClientBuilder contentResponseOnWriteEnabled(boolean contentResponse
         return this;
     }
 
+    /**
+     * Sets the GATEWAY connection configuration to be used.
+     *
+     * @param gatewayConnectionConfig GATEWAY connection configuration
+     * @return current CosmosClientBuilder
+     */
+    public CosmosClientBuilder connectionModeGateway(GatewayConnectionConfig gatewayConnectionConfig) {
+        this.gatewayConnectionConfig = gatewayConnectionConfig;
+        return this;
+    }
+
+    /**
+     * Sets the DIRECT connection configuration to be used.
+     *
+     * @param directConnectionConfig DIRECT connection configuration
+     * @return current CosmosClientBuilder
+     */
+    public CosmosClientBuilder connectionModeDirect(DirectConnectionConfig directConnectionConfig) {
+        this.directConnectionConfig = directConnectionConfig;
+        return this;
+    }
+
+    /**
+     * Gets the GATEWAY connection configuration to be used.
+     *
+     * @return gateway connection config
+     */
+    public GatewayConnectionConfig getGatewayConnectionConfig() {
+        return gatewayConnectionConfig;
+    }
+
+    /**
+     * Gets the DIRECT connection configuration to be used.
+     *
+     * @return direct connection config
+     */
+    public DirectConnectionConfig getDirectConnectionConfig() {
+        return directConnectionConfig;
+    }
```

Comment:
ditto

Reply:
Good catch @moderakh - will get rid of them.